### PR TITLE
f-metadata@2.2.0: promisify and unit tests

### PR DESCRIPTION
**Pull request by asherstoppard**
Promisify f-metadata and add unit tests

## UI Review Checks

- [x] README and/or UI Documentation has been [created|updated]
- [x] Unit tests have been [created|updated]

### CODE REVIEW

File: packages/f-metadata/package.json [hunk 1]
```diff
@@ -25,8 +25,10 @@
     "fozzie"
   ],
   "scripts": {
+    "prepublish": "yarn test",
```
Comment (ashleynolan) [inline packages/f-metadata/package.json]: Can you make this `prepublishOnly`? (otherwise it runs on `npm install` which clogs the build)

Comment (asherstoppard) [reply packages/f-metadata/package.json]: Ah ok, yeah of course :+1:

File: packages/f-metadata/package.json [hunk 2]
```diff
@@ -36,6 +38,7 @@
   },
   "devDependencies": {
     "@vue/cli-plugin-babel": "3.9.2",
-    "@vue/cli-plugin-eslint": "3.9.2"
+    "@vue/cli-plugin-eslint": "3.9.2",
+    "jest": "^25.1.0"
```
Comment (DamianMullins) [inline packages/f-metadata/package.json]: Can you pin this one, please?

File: packages/f-metadata/package.json [hunk 3]
```diff
@@ -25,8 +25,10 @@
     "fozzie"
   ],
   "scripts": {
+    "prepublish": "yarn test",
     "build": "yarn lint",
-    "lint": "vue-cli-service lint --fix"
+    "lint": "vue-cli-service lint --fix",
```
Comment (DamianMullins) [inline packages/f-metadata/package.json]: Might be worth splitting this task out like we do in other repos into `lint` and `lint:fix` tasks?

File: packages/f-metadata/package.json [hunk 4]
```diff
@@ -25,8 +25,10 @@
...
+    "test": "jest --coverage"
```
Comment (ashleynolan) [inline packages/f-metadata/package.json]: Can you use the `vue-cli-service test:unit` here (to keep it consistent with other packages?).

Should be able to set the `collectCoverage` setting then to get coverage out of the tests

Comment (asherstoppard) [reply packages/f-metadata/package.json]: Would we want to couple this module with `vue-cli-service` given that it's pure js? Appreciate there's commands in there already but I'm intrigued as to what it gives us over using the jest/eslint binaries directly?

Comment (ashleynolan) [reply packages/f-metadata/package.json]: See what you mean – I'm not super precious about what's used (as it's all jest ultimately and it's only a dev dependency). I was mainly only thinking about it being an additional monorepo dependency (as `vue-cli-service` is effectively already available through the root depenencies), but happy to directly use jest if you think it fits better here.

**Review by kevinrodrigues — DISMISSED** [review]

File: packages/f-metadata/package.json (hunk 4 above)
Comment (asherstoppard) [reply packages/f-metadata/package.json]: I guess it's more a question for the broader group in the UI guild. I'll change it over to `vue-cli-service` for the purpose of this PR and make note to raise it as a query in the next meeting :+1:

Comment (DamianMullins) [reply packages/f-metadata/package.json]: I guess if `vue-cli-service` provides us the ability to with Jest already then it's one less dependecy to the project. On the other hand having Jest as a dependency means that we can stay up to date with the latest versions without having to wait for vue CLI to update. I'm happy either way. 

We do run tests using Jest directly in other modules as we're not always using the vue CLI.

Comment (ashleynolan) [reply packages/f-metadata/package.json]: Yeh, it's much of a muchness I guess. As @asherstoppard mentioned, as long as we have clear guidance on this as a group it shouldn't really be an issue either way (so we're happy to use more direct dependencies if a module is more native JS, like this one is, but use `vue-cli` for vue-specific components).

**Review by ashleynolan — APPROVED** [review]

**Review by kevinrodrigues — APPROVED** [review]

**Pull request merged**